# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: 62480de389ff
# Full Hash: 62480de389ffba353998de24760993b2e1531232
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth. CLOSED TREE
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r f2e486f1be17 -r 62480de389ff js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Thu Aug 27 10:30:46 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Thu Aug 27 10:53:04 2020 +0000
@@ -233,7 +233,7 @@
   void* pc = unwindState.pc;
   const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->instance()->code();
+  const wasm::Code& code = wasm::GetNearestEffectiveTls(fp)->instance->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the